src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx

```diff
@@ -28,16 +28,26 @@ interface FormattedPrices {
 
 const LOW_PRICE_FLOOR = new BigNumber('0.0001')
 
-export function formatPriceToPrecision(price: BigNumber, useThreshold = false): string {
+interface FormatPriceOptions {
+  useThreshold?: boolean | undefined
+  decimals?: number | undefined
+}
+
+export function formatPriceWithFloor(
+  price: BigNumber,
+  { useThreshold = false, decimals = PRICE_ESTIMATION_PRECISION }: FormatPriceOptions,
+): string {
+  const priceAsBN = parseAmount(price.toString(10), 18)
   return price.gt(LOW_PRICE_FLOOR) || !useThreshold
-    ? price.toFixed(PRICE_ESTIMATION_PRECISION)
+    ? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      formatAmount({ amount: priceAsBN!, precision: 18, decimals })
```

Comment:
by the way, I made a post-merge review of this part, since it was in an unreviewed merge, you haven't seen the mail/notification?

Reply:
reviewed